cpp/include/cudf/strings/string.cuh

```diff
@@ -105,6 +105,30 @@ __device__ bool is_float(string_view const& d_str)
   }
   return result;
 }
+
+inline __device__ string_view shuffle_xor(string_view var, uint32_t delta)
+{
+  const char* data = reinterpret_cast<const char*>(
+    __shfl_xor_sync(~0, reinterpret_cast<uintptr_t>(var.data()), delta));
```

Comment:
Does this exist because you want to use shfl intrinsics on `string_view` members because a `string_view` itself might not fit in the payload size? If so, I'm confused how this is working because I didn't know `__shfl_xor_sync` worked on 64 bit integers which is what `uintptr_t` would be here.

Reply:
It does work with 64 bit data types. From [here](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#synopsis) :

`T __shfl_xor_sync(unsigned mask, T var, int laneMask, int width=warpSize);`

`T` can be `int`, `unsigned int`, `long`, `unsigned long`, `long long`, `unsigned long long`, `float` or `double`. With the `cuda_fp16.h` header included, `T` can also be `__half` or `__half2`. Similarly, with the `cuda_bf16.h` header included, `T` can also be `__nv_bfloat16` or `__nv_bfloat162`.